v3/src/models/data/formula-manager.ts

```diff
@@ -270,7 +272,31 @@ export class FormulaManager {
           this.recalculateFormula(formulaId)
         }
       })
-    }, { equals: comparer.structural, fireImmediately: true })
+    }, {
+      equals: comparer.structural,
+      fireImmediately: true,
+      name: "FormulaManager.registerAllFormulas.reaction"
+    })
+  }
+
+  unregisterDeletedFormulas() {
+    this.formulaMetadata.forEach((metadata, formulaId) => {
+      const { dataSetId, attributeId } = this.getFormulaMetadata(formulaId)
+      const dataSet = this.dataSets.get(dataSetId)
+      if (!dataSet) {
+        this.unregisterFormula(formulaId)
+        return
+      }
+      if (!dataSet.attrFromID(attributeId)) {
+        this.unregisterFormula(formulaId)
+        return
+      }
```

Comment:
Seems like this could be combined to:
```suggestion
      if (!dataSet?.attrFromID(attributeId)) {
        this.unregisterFormula(formulaId)
        return
      }
```